### PR TITLE
feat: enable linters and fix errors

### PR DESCRIPTION
**Pull request by danroc**
This PR enables linter on `.tsx` files and fixes all detected errors. Please note that some small refactors were required to properly fix some of the problems.

### CODE REVIEW

File: packages/site/src/types/custom.d.ts [hunk 1]
```diff
@@ -1,13 +1,13 @@
-/* eslint-disable*/
-
+// eslint-disable-next-line spaced-comment
 /// <reference types="react-scripts" />
```
Comment (gantunesr) [inline packages/site/src/types/custom.d.ts:2]: Can't we remove this comment?

**Review by gantunesr — DISMISSED** [review]
lgtm

File: package.json [hunk 2]
```diff
@@ -26,11 +26,12 @@
     "build": "yarn workspaces foreach --parallel --topological --verbose run build",
     "build:dapp": "yarn workspace @metamask/snap-simple-keyring-site build",
     "build:snap": "yarn workspace @metamask/snap-simple-keyring-snap build",
-    "lint": "yarn lint:eslint && yarn lint:misc --check && yarn lint:deps",
+    "lint": "yarn lint:eslint && yarn lint:misc --check && yarn lint:deps && yarn lint:types",
```
Comment (HowardBraham) [inline package.json:29]: We should discuss whether we actually want this line to be

```suggestion
    "lint": "yarn workspaces foreach --parallel run lint",
```

Right now, we're running a weird hybrid lint, where `lint:eslint` and `lint:misc` are running at the root of the repo, but `lint:deps` and `lint:types` are running in each workspace.

File: packages/site/src/types/custom.d.ts (hunk 1 above)
Comment (danroc) [reply packages/site/src/types/custom.d.ts:2]: If we remove this line, eslint will replace

```ts
/// <reference types="react-scripts" />
```

with

```ts
// / <reference types="react-scripts" />
```

But I just found a rule to change this behaviour, I'll try it.

File: package.json (hunk 2 above)
Comment (danroc) [reply package.json]: Good catch! I replaced the root targets by `workspaces foreach` ones.

File: packages/site/src/types/custom.d.ts (hunk 1 above)
Comment (danroc) [reply packages/site/src/types/custom.d.ts]: Fixed 🎉

**Review by montelaidev — APPROVED** [review]

**Pull request merged**